emeraldtree/html.py

```diff
@@ -96,14 +96,10 @@ def handle_starttag(self, tag, attrs):
                 elif k == "content":
                     content = v
             if http_equiv == "content-type" and content:
-                # use mimetools to parse the http header
-                import mimetools, StringIO
-                header = mimetools.Message(
-                    StringIO.StringIO("%s: %s\n\n" % (http_equiv, content))
-                    )
-                encoding = header.getparam("charset")
-                if encoding:
-                    self.encoding = encoding
+                import cgi
+                _, params = cgi.parse_header(content)
+                if params.get('charset'):
+                    self.encoding = params['charset']
```

Comment:
```
encoding = params.get('charset')
if encoding:
    self.encoding = encoding
```
This is a smaller diff and also helps for defect headers like `text/plain;charset=`.

Reply:
No problem, incoming...

Reply:
All done, and you're right, it does look nicer.